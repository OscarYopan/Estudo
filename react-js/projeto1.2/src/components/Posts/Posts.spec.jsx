import { Posts } from '.'
const { render,screen } = require("@testing-library/react")

const props = {
  posts: [
    {
      id: 1,
      title: 'Title One',
      body: 'Full Body Two',
      cover: 'img/img.png'
    },
    {
      id: 2,
      title: 'Title Two',
      body: 'Full Body Two',
      cover: 'img/img2.png'
    },
    {
      id: 3,
      title: 'Title Three',
      body: 'Full Body Three',
      cover: 'img/img3.png'
    },
  ]
};

describe('<Posts />', () => {
  it('should render Posts', () => {
    render(<Posts {...props} />)

    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3)

    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3)

    expect(screen.getAllByText(/body/i)).toHaveLength(3)

    expect(screen.getByRole('img', { name: /Title Three/i })).toHaveAttribute('src', 'img/img3.png')
  })
})