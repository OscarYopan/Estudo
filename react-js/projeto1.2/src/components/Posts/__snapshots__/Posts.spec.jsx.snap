// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title One"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title One
      </h1>
      <p>
        Full Body Two
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title Two"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title Two
      </h1>
      <p>
        Full Body Two
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title Three"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title Three
      </h1>
      <p>
        Full Body Three
      </p>
    </div>
  </div>
</div>
`;

exports[`<Posts /> should match snapshot 2`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title One"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title One
      </h1>
      <p>
        Full Body Two
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title Two"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title Two
      </h1>
      <p>
        Full Body Two
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title Three"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title Three
      </h1>
      <p>
        Full Body Three
      </p>
    </div>
  </div>
</div>
`;

exports[`<Posts /> should not render posts 1`] = `
<div
  class="posts"
/>
`;
